commandbased/flowcontrol.py

```diff
@@ -19,7 +19,11 @@ def _getCommandGroup():
     '''
     # https://stackoverflow.com/a/14694234
     stack = inspect.stack()
-    frame = stack[2].frame
+    try:
+        frame = stack[2].frame
+    except AttributeError:
+        # Python < 3.5
+        frame = stack[2][0]
```

Comment:
`inspect.stack()` returns a list of named tuples, so `stack[2][0]` should also work on Python 3.5+. However, `inspect.currentframe().f_back.f_back` or even `sys._getframe(2)` should work as well. Best to avoid unnecessary work.

I wonder if there's a better way of doing all this though. Probably not.

Reply:
Unfortunately, both `sys._getframe` and `inspect.currentframe` are implementation dependent. It's a shame to lose readability just to maintain compatibility with 2014, but it makes more sense than keeping a check just to point out that we're doing that.